src/components/comments.js

```diff
@@ -1,33 +1,36 @@
-import React from 'react';
+import React from "react";
 
 import { StaticQuery, graphql } from "gatsby";
 
-import { Disqus } from 'gatsby-plugin-disqus'
+import { Disqus } from "gatsby-plugin-disqus";
 
 function Comments({ data, title, location }) {
-  const disqusConfig = {
-    url: `${data.site.siteMetadata.siteUrl + location}`,
-    identifier: title,
-    title: title,
-  }
-  return (
-    <Disqus config={disqusConfig} />
-  )
+    if (data.site.siteMetadata.disqus === null) return <></>;
+
+    const disqusConfig = {
+        url: `${data.site.siteMetadata.siteUrl + location}`,
+        identifier: title,
+        title: title
+    };
+    return <Disqus config={disqusConfig} />;
 }
 
 export default ({ title, location }) => {
-  return (
-    <StaticQuery
-      query={graphql`
-            query {
-                site {
-                    siteMetadata {
-                        siteUrl
+    return (
+        <StaticQuery
+            query={graphql`
+                query {
+                    site {
+                        siteMetadata {
+                            siteUrl
+                            disqus
+                        }
                     }
                 }
-            }
-        `}
-      render={data => <Comments data={data} title={title} location={location} />}
-    />
-  )
-}
\ No newline at end of file
+            `}
+            render={data => (
+                <Comments data={data} title={title} location={location} />
```

Comment:
i think we can simplify the props here by only giving the url it needs.
something like this
```javascript
render={(data) => {
const url = data.site.siteMetadata.siteUrl + location;
return <Comments url={url} title={title} />
}}
```

Reply:
Yeah that would look more pretty I guess.